griptape/tools/aws_iam_client/tool.py

```diff
@@ -1,20 +1,15 @@
 from __future__ import annotations
-from typing import TYPE_CHECKING, Optional
 from schema import Schema, Literal
 from attr import define, field, Factory
 from griptape.artifacts import TextArtifact, ErrorArtifact, ListArtifact
 from griptape.utils.decorators import activity
 from griptape.tools import BaseAwsClient
-
-if TYPE_CHECKING:
-    import boto3
+from mypy_boto3_iam import Client
```

Comment:
1. Are we assuming `TYPE_CHECKING` is always True?
2. Is anything different between `mypy_boto3_iam` and the "real" `boto3`?